src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/InitializeOperation.cs

```diff
@@ -34,8 +40,8 @@ protected override async Task ExecuteInternalAsync(
         CancellationToken cancellationToken)
         => await _vmrInitializer.InitializeRepository(
             repoName,
-            targetRevision,
-            null,
+            (await _barClient.GetBuildsAsync(repoName, targetRevision)).SingleOrDefault()
+                ?? throw new Exception($"Failed to, or found multiple builds for {repoName}/{targetRevision}"),
```

Comment:
The message seems a bit weird to be honest. I would rather rephrase it to something like "Expected 1 build to exist for commit X of repo Y" or similar.